[FEATURE] ENUM Type, Status 정의

donsonioc2010 opened the pull request:
## 📝 PR Summary

Core-Enum에 활용할 종류들 사전 정의

#### 🌲 Working Branch

feat/#13

#### 🌲 TODOs
- QNA
    - [x] AnswerStatus (답변 상태)
    - [x] QnaProgressStatus (QNA 진행상태)
- Song
    - [x] MachineType (회사 기계 종류)
- User
    - [ ] RegisterType (Auth 가입 회사 종류)
    - [x] UserGender (성별)
    - [x] UserRole (유저 권한)
    - [x] UserStatus (계정 상태) 

### Related Issues

- #13 

### 📚 Remarks
과거 읽었던 이동욱님의 글을 보고 enum을 해당 방식으로 사용을 많이 해보고 있습니다 ㅎ...
https://techblog.woowahan.com/2527/

컨트롤러를 추가해놔서 아래의 mapping으로 모든 enum확인이 가능하게 해놨습니다.
- http://localhost:8080/enum
- http://localhost:8080/enum/{key}
    - ex)http://localhost:8080/enum/QnaProgressStatus

donsonioc2010 committed:
feat : Enum정의

- 기존 Enum 모두 core-enum이동
- 현재 enum으로 사용할 목록들 정의
- Enum의 목록 호출 컨트롤러 생성

JakePark929 commented on core/core-enum/src/main/java/com/nerd/favorite18/core/enums/user/RegisterType.java:
```diff
@@ -0,0 +1,23 @@
+    public String getValue() {
+        return value;
+    }
+}
```
6/18 DB 피드백 중에 이 부분은 나누지 않기로 해서 제외해도 될 것 같습니다 :)

JakePark929 commented on core/core-enum/src/main/java/com/nerd/favorite18/core/enums/user/UserGender.java:
```diff
@@ -0,0 +1,22 @@
+
+@RequiredArgsConstructor
+public enum UserGender implements EnumMapperType {
+    MEN("남자"),
```
이 부분 오타가 있었는데 그대로 들어갔네요 😥
번거로우시겠지만 수정 부탁드리겠습니다.

JakePark929 commented on core/core-enum/src/main/java/com/nerd/favorite18/core/enums/qna/QnaProgressStatus.java:
```diff
@@ -0,0 +1,25 @@
+    COMPLETED("처리완료"),
+    CLOSED("닫힘"),
+    ON_HOLD("보류중");
+
```
이 부분은 해석하기에 따라 다른 의미를 가질 수 있어서 주석으로 상황을 명확하게 알려 주시면 도움 될 것 같습니다!

donsonioc2010 replied:
CLOSED는 불필요하다 느껴서 삭제하고, 나머지는 각 상태별로 주석들 추가는 해놨습니다.

추후 개발하면서 변경 가능합니다 :)

JakePark929 commented:
예전에 이동욱 님 Enum에 관한 글 잠깐 봤었는데 그 땐 이해가 잘 안 갔는데, 
지금 보니 아주 좋은 활용 방법이네요! 자주 써야 겠습니다. :D

코멘트 사항 확인 해주시고, 어프루브 진행하겠습니다~

donsonioc2010 committed:
Merge remote-tracking branch 'refs/remotes/origin/dev' into feat/#13

# Conflicts:
#	storage/db-core/src/main/java/com/nerd/favorite18/storage/db/core/user/type/UserGender.java

JakePark929 reviewed (commented):
예전에 이동욱 님 Enum에 관한 글 잠깐 봤었는데 그 땐 이해가 잘 안 갔는데,
지금 보니 아주 좋은 활용 방법이네요! 자주 써야 겠습니다. :D

코멘트 사항 확인 해주시고, 어프루브 진행하겠습니다~

donsonioc2010 committed:
chore : GitIgnore 추가

donsonioc2010 committed:
fix : 패키지 경로가 프로젝트와 일치하지 않은 부분 수정.

donsonioc2010 committed:
feat : Enum정의 추가

- RegisterType의 경우 이미 SubId에 존재하여 삭제 및 그에 따른 Config변경
- UserGender의 경우 오탈자가 존재하여 수정
- QnaProgressStatus의 각 상태에 대하여, 설명 추가
- EnumController에서 사용하던 패키지 경로변경으로 인한 수정.
- WebConfig에서 enumController경로 추가.

JakePark929 commented on core/core-enum/src/main/java/com/nerd/favorite18/core/enums/user/UserGender.java:
```diff
@@ -0,0 +1,22 @@
+
+@RequiredArgsConstructor
+public enum UserGender implements EnumMapperType {
+    MAN("남자"),
```
👍

JakePark929 commented on core/core-api/src/main/java/com/nerd/favorite18/core/api/_common/config/EnumCodeConfig.java:
```diff
@@ -0,0 +1,34 @@
+
+@Configuration
+public class EnumCodeConfig {
+    @Bean
```
👍

JakePark929 reviewed (approved):
LGTM 👍
